### ISSUE TITLE
GOME 40

### ISSUE THREAD

**Issue by Manlibear** [mzur/gnome-shell-always-indicator#9]
Do you have any plans to update this extension to GNOME 40?

**Comment by mzur** [comment]
Not yet, as I'm still using 3.36. Are you willing to help? As a first step, you could look for error messages of this extension in `journalctl /usr/bin/gnome-shell`.

**Comment by Manlibear** [comment]
I can certainly have a look into it, haven't done any GNOME dev work before, but gotta start sometime! I can't even get the extension loaded, doesn't show up under Extensions.

**Comment by mzur** [comment]
That's because GNOME 40 does a strict version check. Extensions which do not explicitly specify compatibility are not loaded. Try changing the [supported version](https://github.com/mzur/gnome-shell-always-indicator/blob/d2e5c6f840d397a8d423d05ac3f86b68a2f785c7/always-indicator%40martin.zurowietz.de/metadata.json#L6)  in the `metadata.json` to 40.0.

**Comment by Manlibear** [comment]
I had tried that without success, spent a little time comparing your source with a 40.0 extension. I think settings dialogs need to be GTK4 now (don't quote me though, think I read about it on the GNOME blog), so maybe that could prevent it loading?

**Comment by mzur** [comment]
That could be it. If anyone wants to give it a try, check out [this guide](https://gjs.guide/extensions/upgrading/gnome-shell-40.html). Also, please leave a note here.

**Comment by v4rakh** [comment]
Hi. The extension seems to work just fine from functionality perspective, but as @Manlibear stated, the `Settings.ui` needs to be ported to GTK4.

Unfortunately I also don't have any experience regarding GNOME extensions development and currently no time to dive into it, but there's a migration guide for `.ui` files stated in the guide.

When trying to open the settings panel, the following shows up:

```
Gtk.BuilderError: /home/varakh/.local/share/gnome-shell/extensions/always-indicator@martin.zurowietz.de/settings.ui:9:35 Invalid property: GtkBox.margin_left

Stack trace:
  _init@/home/varakh/.local/share/gnome-shell/extensions/always-indicator@martin.zurowietz.de/prefs.js:17:23
  wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
  buildPrefsWidget@/home/varakh/.local/share/gnome-shell/extensions/always-indicator@martin.zurowietz.de/prefs.js:59:17
  _init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:209:40
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
  run@resource:///org/gnome/gjs/modules/script/package.js:222:19
  start@resource:///org/gnome/gjs/modules/script/package.js:206:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
  
```

**Comment by mzur** [comment]
I just released v3.0 which is compatible with GNOME 40. It will be published on extensions.gnome.org soon.
